### ISSUE TITLE
`HtmlElementView` inside `ClipRRect` doesn't work properly in Safari with CanvasKit

### ISSUE THREAD

**Issue by daohoangson** [flutter/flutter#91191]
Apparently ClipRRect + Safari + CanvasKit is not a good combination.

| Test | Renderer | Browser | Pass/fail |
| --- | --- | --- | --- |
| HtmlElementView only | html | Chrome | ✅ |
| HtmlElementView only | html | Safari | ✅ |
| HtmlElementView only | canvaskit | Chrome | ✅ |
| HtmlElementView only | canvaskit | Safari | ✅ |
| HtmlElementView inside ClipRRect | html | Chrome | ✅ |
| HtmlElementView inside ClipRRect | html | Safari | ✅ |
| HtmlElementView inside ClipRRect | canvaskit | Chrome | ✅ |
| HtmlElementView inside ClipRRect | canvaskit | Safari | ❌ |

## Steps to Reproduce

1. Run `flutter create bug`.
2. Update the files as follows: 

### pubspec.yaml

```yaml
name: bug
description: A new Flutter project.
publish_to: 'none'
version: 1.0.0+1

environment:
  sdk: ">=2.12.0 <3.0.0"

dependencies: 
  flutter:
    sdk: flutter
  fwfh_webview: ^0.6.1+2

dev_dependencies:
  flutter_lints: ^1.0.0

flutter:
  uses-material-design: true
```

### main.dart

```dart
import 'dart:html';

import 'package:flutter/material.dart';
import 'package:fwfh_webview/src/external/dart_ui.dart' as ui;

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      home: Scaffold(
        appBar: AppBar(
          title: const Text('Flutter Demo'),
        ),
        body: ClipRRect(
          borderRadius: BorderRadius.circular(10),
          child: _TestWidget(),
        ),
      ),
    );
  }
}

class _TestWidget extends StatefulWidget {
  @override
  State<_TestWidget> createState() => _TestWidgetState();
}

class _TestWidgetState extends State<_TestWidget> {
  final _element = DivElement();
  late Widget _htmlElementView;

  @override
  void initState() {
    super.initState();

    _element.style.background = 'red';

    final viewType = '$this#$hashCode';
    ui.platformViewRegistry.registerViewFactory(viewType, (_) => _element);
    _htmlElementView = HtmlElementView(viewType: viewType);
  }

  @override
  Widget build(BuildContext context) => _htmlElementView;
}
```

3. Run the app with `flutter run -d chrome --web-renderer canvaskit`. Open the same URL in Safari.

**Expected results:** (in Chrome) 

<img width="981" alt="Screen Shot 2021-10-03 at 20 45 36" src="https://user-images.githubusercontent.com/239336/135756318-a2f1ee09-3858-4dd2-bce9-f7b00522bc90.png">

**Actual results:** (in Safari) 

<img width="981" alt="Screen Shot 2021-10-03 at 20 45 50" src="https://user-images.githubusercontent.com/239336/135756323-606ea1f1-f3b2-44c5-8b51-0a64a58f6dbf.png">

<details>
  <summary>Logs</summary>

```
[✓] Flutter (Channel master, 2.6.0-12.0.pre.185, on macOS 11.3 20E232 darwin-x64, locale en-VN)
    • Flutter version 2.6.0-12.0.pre.185 at /Users/sondh/fvm/versions/master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 91dd3276fd (21 hours ago), 2021-10-02 10:18:03 -0700
    • Engine revision 6f4143e800
    • Dart version 2.15.0 (build 2.15.0-172.0.dev)

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/sondh/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[✓] Xcode - develop for iOS and macOS (Xcode 13.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Ultimate Edition (version 2021.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.60.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.26.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 11.3 20E232 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 94.0.4606.61

! Doctor found issues in 1 category.
```

</details>

**Comment by danagbemava-nc** [comment]
@daohoangson, thanks for the detailed report. 

Issue is reproducible on `master` and `stable` using sample provided by OP above. 
It occurs only with Safari as stated by OP and the issue only reproduces with `canvaskit`. 

<details>
<summary>flutter doctor -v</summary>

```console
[✓] Flutter (Channel stable, 2.5.2, on macOS 11.5.1 20G80 darwin-arm, locale en-GB)
    • Flutter version 2.5.2 at /Users/nexus/dev/sdks/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 3595343e20 (4 days ago), 2021-09-30 12:58:18 -0700
    • Engine revision 6ac856380f
    • Dart version 2.14.3

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.60.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.26.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 11.5.1 20G80 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 94.0.4606.61

• No issues found!
```

```console
[✓] Flutter (Channel master, 2.6.0-12.0.pre.185, on macOS 11.5.1 20G80 darwin-arm, locale en-GB)
    • Flutter version 2.6.0-12.0.pre.185 at /Users/nexus/dev/sdks/flutters
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 91dd3276fd (2 days ago), 2021-10-02 10:18:03 -0700
    • Engine revision 6f4143e800
    • Dart version 2.15.0 (build 2.15.0-172.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 12.5.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.0.0.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.60.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.26.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 11.5.1 20G80 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 94.0.4606.61

! Doctor found issues in 1 category.
```
</details>

<details>
<summary>screenshots</summary>

| html | canvaskit|
|----- | ---------|
| <img width="788" alt="Screenshot 2021-10-04 at 10 10 19" src="https://user-images.githubusercontent.com/88313112/135837236-a60071f2-d86c-4752-b8e0-8405b41bc024.png">| <img width="788" alt="Screenshot 2021-10-04 at 10 02 55" src="https://user-images.githubusercontent.com/88313112/135837263-6d46f4fa-00d3-45a2-bd27-cc306f3701df.png"> |


</details>

**Comment by rignaneseleo** [comment]
any fix/workaround to this?

**Comment by zvikarp** [comment]
@danagbemava-nc  I can confirm this issue still exists in flutter 3.3.7-stable, even though this issue was opened over a year ago. it is important that flutter can run smoothly on all popular browsers.

No one else is testing their apps on safari + canvaskit? or there is a workaround I'm not aware of?

**Comment by daohoangson** [comment]
It looks like 3.3.10 has fixed this bug when it was trying to fix #80401. I'm closing the issue now.
